compiler/rustc_target/src/target_features.rs

```diff
@@ -582,6 +582,12 @@ pub fn all_rust_features() -> impl Iterator<Item = (&'static str, Stability)> {
 const X86_FEATURES_FOR_CORRECT_VECTOR_ABI: &'static [(u64, &'static str)] =
     &[(128, "sse"), (256, "avx"), (512, "avx512f")];
 const AARCH64_FEATURES_FOR_CORRECT_VECTOR_ABI: &'static [(u64, &'static str)] = &[(128, "neon")];
+const POWERPC_FEATURES_FOR_CORRECT_VECTOR_ABI: &'static [(u64, &'static str)] = &[(128, "altivec")];
+const RISCV_FEATURES_FOR_CORRECT_VECTOR_ABI: &'static [(u64, &'static str)] =
+    &[(32, "zve32x"), (64, "zve64x"), (128, "zvl128b")];
+const WASM_FEATURES_FOR_CORRECT_VECTOR_ABI: &'static [(u64, &'static str)] = &[(128, "simd128")];
+const S390X_FEATURES_FOR_CORRECT_VECTOR_ABI: &'static [(u64, &'static str)] = &[(128, "vector")];
+const SPARC_FEATURES_FOR_CORRECT_VECTOR_ABI: &'static [(u64, &'static str)] = &[(64, "v9")];
```

Comment:
AFAIK, this should be gated by vis, not by v9.

Reply:
(vis is not included in SPARC_FEATURES, though)

Reply:
Given that, I think we should belay changing SPARC.

Reply:
We do have v9, and if that implies vis then it should be fine for now?

We should have a FIXME to consider weakening this to `vis` once that is a supported target feature.

Reply:
Do you mean "never warn on SPARC" or "always warn on SPARC"?
I am not sure there's any way to use VIS vector types / instructions correctly in rust (with C ABIs, at least) nowadays if the "vis" feature cannot be enabled...

Reply:
@RalfJung 

> We do have v9, and if that implies vis then it should be fine for now?

vis is an optional extension of v9 ISA, so v9 doesn't imply vis.

---

@veluca93 

I think it is okay to "always warn". The only vector types currently available for this target are core::simd and custom repr(simd) types, which are unstable. We can modify the conditions here when adding the vis target feature and the corresponding core::arch type.

Reply:
Done. (also for riscv)